src/lib.rs

```diff
@@ -7,9 +7,10 @@
 // at your option. This file may not be copied, modified, or distributed except according to those
 // terms.
 
-//! An immutable cactus stuck (also called a spaghetti stack or parent pointer tree). A cactus
-//! stack is a (possibly empty) node with a (possibly null) pointer to a parent node. Any given
-//! node has a unique path back to the root node. Rather than mutably updating the stack, one
+//! An immutable parent pointer tree (also called a cactus stack or spaghetti stack).
+//!
+//! A cactus stack is a (possibly empty) node with a (possibly null) pointer to a parent node. Any
```

Comment:
Wait. Something isn't right. We've got mixed terminology again.

You wanted "immutable parent pointer" across the board right?

It's also just occurred to me that the package name is "cactus". Should it be "cactus stack" across the board? Sorry for not noticing earlier.

Reply:
I think most people call these "parent pointer trees", so I wanted that to be the first description people see, otherwise they might not realise what the crate does. Something like "cactus: a parent pointer tree". But after that I think it's fine to call it a "cactus stack" (because that's the crate name).

Reply:
So if I understand correctly, you want to use "cactus stack" across the board, but drop in a mention of the alternative names for clarity. So you want (?):

> A cactus stack (also called an immutable parent pointer tree or a spaghetti stack).
>
> A cactus stack is ...

Reply:
"Parent pointer tree" would be the most widely understood name for the crate, but it's way too long. "Cactus" is a cute name, and nice and short, but less widely known. So I wanted to say "Here's the name you know it by. Here's what it's going to be called from now on."

Reply:
I see. So no changes required? Shall I merge?

Reply:
Well, I think you've convinced me it's confusing, at least a little bit. See if that new commit works for you a bit better.